modules/stochastic_tools/include/transfers/SamplerTransfer.h

```diff
@@ -0,0 +1,61 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef SAMPLERTRANSFER_H
+#define SAMPLERTRANSFER_H
+
+// MOOSE includes
+#include "MultiAppTransfer.h"
+#include "Sampler.h"
+
+// Forward declarations
+class SamplerTransfer;
+class SamplerReceiver;
+
+template <>
+InputParameters validParams<SamplerTransfer>();
+
+/**
+ * Copy each row from each DenseMatrix to the sub-applications SamplerReceiver object.
+ */
+class SamplerTransfer : public MultiAppTransfer
+{
+public:
+  SamplerTransfer(const InputParameters & parameters);
+  virtual void execute() override;
+
+protected:
+  /**
+   * Return the SamplerReceiver object and perform error checking.
+   * @param app_index The global sup-app index
+   *
+   * This method is simply for keeping the execute() method clean.
+   */
+  std::shared_ptr<SamplerReceiver> getReceiver(unsigned int app_index,
```

Comment:
You aren't transferring or sharing ownership here so this is correct, semantically. The warehouse retains ownership of Controls for the duration of the simulation. I would much prefer to see a raw pointer returned here to represent sharing of data, not ownership.